node/bft/src/gateway.rs

```diff
@@ -1412,8 +1413,14 @@ impl<N: Network> Gateway<N> {
 
         // Sample a random nonce.
         let our_nonce = rng.r#gen();
+        // Determine the snarkOS SHA to send to the peer.
+        let current_block_height = self.ledger.latest_block_height();
+        let consensus_version = N::CONSENSUS_VERSION(current_block_height).unwrap();
+        let snarkos_sha = (consensus_version >= ConsensusVersion::V12)
+            .then(|| built_info::GIT_COMMIT_HASH.unwrap_or_default().into());
```

Comment:
nit: wouldn't this result in `Some("")` in case `GIT_COMMIT_HASH` is missing (shouldn't happen)?